aspnetcore/blazor/fundamentals/startup.md

```diff
@@ -0,0 +1,122 @@
+uid: blazor/fundamentals/startup
+---
+# ASP.NET Core Blazor Startup
+
+Configure a manual start in the `wwwroot/index.html` file (Blazor WebAssembly) or `Pages/_Host.cshtml` file (Blazor Server):
+
+* Add an `autostart="false"` attribute and value to the `<script>` tag for the Blazor script.
+* Place a script that calls `Blazor.start` after the Blazor `<script>` tag and inside the closing `</body>` tag.
+
+## Initialize Blazor when the document is ready
+
+The following example starts Blazor when the document is ready:
+
+```cshtml
+<body>
+    ...
+
+    <script src="_framework/blazor.{webassembly|server}.js" autostart="false"></script>
+    <script>
+      document.addEventListener("DOMContentLoaded", function() {
+        Blazor.start();
+      });
+    </script>
+</body>
+```
+
+The `{webassembly|server}` placeholder in the preceding markup is either `webassembly` for a Blazor WebAssembly app (`blazor.webassembly.js`) or `server` for a Blazor Server app (`blazor.server.js`).
+
+## Chain to the `Promise` that results from a manual start
+
+To perform additional tasks, such as JS interop initialization, use [`then`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise/then) to chain to the [`Promise`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise) that results from a manual Blazor app start:
+
+```cshtml
+<body>
+    ...
+
+    <script src="_framework/blazor.{webassembly|server}.js" autostart="false"></script>
+    <script>
+      Blazor.start().then(function () {
+        ...
+      });
+    </script>
+</body>
+```
+
+The `{webassembly|server}` placeholder in the preceding markup is either `webassembly` for a Blazor WebAssembly app (`blazor.webassembly.js`) or `server` for a Blazor Server app (`blazor.server.js`).
+
+## Load boot resources
+
+Override the built-in boot resource loading mechanism so that boot resources can be fetched from a custom source, such as an external Content Delivery Network (CDN).
+
+* `type`: The type of the resource to load.
+* `name`: The name of the resource to load.
+* `defaultUri`: The URI from which the framework fetches the resource by default. Relative or absolute URIs are supported.
+* `integrity`: The integrity string representing the expected content in the response.
+* `return`: A URI string or a [`Response` promise](https://developer.mozilla.org/docs/Web/API/Response) to override the loading process, or null/undefined to allow the default loading behavior.
+
+```html
+<script src="_framework/blazor.{webassembly|server}.js" autostart="false"></script>
+<script>
+  Blazor.start({
+    loadBootResource: function (type, name, defaultUri, integrity) {
+      ...
+
+      var responseContent = ...;
+      var header_value_1 = ...;
+      var header_value_2 = ...;
+
+      return new Response(responseContent, { 
```

Comment:
This example is a little contrived. Generally, we'd expect the user to do something like issue a fetch call to a CDN or other service here instead of manually creating the response object.

Reply:
> a little contrived

That's by-design at this point. Idk exactly what to show. This new bit cross-links to the only example that we have at ...

https://docs.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-5.0#compression

I'll poke around on the PU repo and Net to see if I can find a real and alternative-to-compression example.

Reply:
*AH, HA!* ... I found DR's blog post for this. I'm going to revise and include all of his text.